ext/WeightInitializersCUDAExt.jl

```diff
@@ -38,6 +39,7 @@ end
 
 function identity_init(rng::AbstractCuRNG, ::Type{T}, dims::Integer...;
         gain::Number=1, shift::Integer=0) where {T <: Number}
+    gain = gain isa T ? gain : convert(T, gain)
```

Comment:
I don't think you need to check for this. The compiler is smart enough to get rid of these

Reply:
should we also get rid of [this](https://github.com/LuxDL/WeightInitializers.jl/blob/main/src/initializers.jl#L59) then?

Reply:
the `T(gain)` specifically

Reply:
sorry scratch that, I misunderstood

Reply:
I meant `gain isa T ? gain : convert(T, gain)` can be written as `T(gain)` and the compiler should be smart to remove it if the types match.

Reply:
ah I see, should I take that approach with all of the checks?